### PR TITLE
Add media bbcode tag

### PR DESCRIPTION
**Pull request by itkg-ngilain**
https://github.com/open-orchestra/open-orchestra-display-bundle/pull/172
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/107
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1051
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/29

### CODE REVIEW

File: MediaBundle/BBcode/AbstractMediaCodeDefinition.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+use OpenOrchestra\Media\Model\MediaInterface;
+
+/**
+ * Class AbstractMediaCodeDefinition
+ */
+abstract class AbstractMediaCodeDefinition extends BBcodeDefinition
+{
+    protected $repository;
+    protected $displayMediaManager;
+    const MEDIA_NOT_FOUND = '<img src="/bundles/openorchestramedia/img/media_not_found.png" style="height:22px;width:22px;margin: 0 3px 0 3px;>';
```
Comment (itkg-alavieille) [inline MediaBundle/BBcode/AbstractMediaCodeDefinition.php]: may be add this in paremeter in configuration

File: MediaBundle/BBcode/AbstractMediaCodeDefinition.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    /**
+     * Get requested media format
+     * 
+     * @param BBcodeElementNodeInterface $el
+     * 
+     * @return string
+     */
+        protected function getFormat(BBcodeElementNodeInterface $el)
```
Comment (itkg-alavieille) [inline MediaBundle/BBcode/AbstractMediaCodeDefinition.php]: indentation

**Comment by itkg-ngilain** [comment]
@itkg-nanne, @itkg-alavieille : check this please

File: MediaBundle/BBcode/MediaCodeDefinition.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+     * Get requested media format
+     * 
+     * @param BBcodeElementNodeInterface $el
+     * 
+     * @return string
+     */
+    protected function getFormat(BBcodeElementNodeInterface $el)
+    {
+        $options = $el->getAttribute();
+        return $options['media'];
```
Comment (itkg-alavieille) [inline MediaBundle/BBcode/MediaCodeDefinition.php]: add an empty line

File: MediaBundle/Resources/config/service.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+parameters:
+    open_orchestra_media.bbcode.definition.media.class: OpenOrchestra\MediaBundle\BBcode\MediaCodeDefinition
+    open_orchestra_media.bbcode.definition.media_without_format.class: OpenOrchestra\MediaBundle\BBcode\MediaWithoutFormatCodeDefinition
+    open_orchestra_media.bbcode.code_definitions.class: OpenOrchestra\MediaBundle\BBcode\MediaBundleBBcodeCollection
+    open_orchestra_media.bbcode.media_not_found: <img src="/bundles/openorchestramedia/img/media_not_found.png" style="height:22px;width:22px;margin: 0 3px 0 3px;>
```
Comment (itkg-alavieille) [inline MediaBundle/Resources/config/service.yml]: i don't agree with the absolute path in parameter

File: MediaBundle/Tests/BBcode/AbstractMediaCodeDefinitionTest.php [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+     * 
+     * @dataProvider provideFormat
+     */
+    public function testGetPreviewHtmlWithGoodMediaId($expectedFormat)
+    {
+        $html = $this->definition->getPreviewHtml($this->BBcodeElementNode);
+
+        Phake::verify($this->displayManager)->displayMediaForWysiwyg($this->media, $expectedFormat);
+    }
+}
```
Comment (itkg-alavieille) [inline MediaBundle/Tests/BBcode/AbstractMediaCodeDefinitionTest.php]: eof

**Comment by itkg-ngilain** [comment]
@itkg-alavieille : I switch the 'not found' html tag from the conf to a specific twig

File: MediaBundle/BBcode/AbstractMediaCodeDefinition.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+namespace OpenOrchestra\MediaBundle\BBcode;
+
+use OpenOrchestra\Media\Repository\MediaRepositoryInterface;
+use OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinition;
+use OpenOrchestra\BBcodeBundle\ElementNode\BBcodeElementNodeInterface;
+use OpenOrchestra\BBcodeBundle\ElementNode\BBcodeElementNode;
+use OpenOrchestra\Media\DisplayMedia\DisplayMediaManager;
+use OpenOrchestra\Media\Model\MediaInterface;
+use Symfony\Component\Templating\EngineInterface;
+/**
```
Comment (itkg-alavieille) [inline MediaBundle/BBcode/AbstractMediaCodeDefinition.php]: add an empty line

File: MediaBundle/BBcode/AbstractMediaCodeDefinition.php [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+abstract class AbstractMediaCodeDefinition extends BBcodeDefinition
+{
+    protected $repository;
+    protected $displayMediaManager;
+    protected $templating;
+
+    /**
+     * @param MediaRepositoryInterface $repository
+     * @param DisplayMediaManager      $displayMediaManager
+     */
```
Comment (itkg-alavieille) [inline MediaBundle/BBcode/AbstractMediaCodeDefinition.php]: update comment

**Comment by itkg-ngilain** [comment]
@itkg-alavieille : update done

File: MediaBundle/Resources/translations/messages.fr.yml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+open_orchestra_media:
+    strategy:
+        video:
+            tag_not_supported: Votre navigateur ne supporte pas la balise video.
```
Comment (itkg-alavieille) [inline MediaBundle/Resources/translations/messages.fr.yml]: it isn't the tag who isn't support it's the format

File: MediaBundle/Tests/BBcode/MediaCodeDefinitionTest.php [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+     * Test usesOption
+     */
+    public function testUsesOption()
+    {
+        $this->assertSame(true, $this->definition->usesOption());
+    }
+
+    /**
+     * Provide expected format
+     */
```
Comment (itkg-alavieille) [inline MediaBundle/Tests/BBcode/MediaCodeDefinitionTest.php]: add comment ``@return array``

File: MediaBundle/Tests/BBcode/MediaWithoutFormatCodeDefinitionTest.php [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+     * Test usesOption
+     */
+    public function testUsesOption()
+    {
+        $this->assertSame(false, $this->definition->usesOption());
+    }
+
+    /**
+     * Provide expected format
+     */
```
Comment (itkg-alavieille) [inline MediaBundle/Tests/BBcode/MediaWithoutFormatCodeDefinitionTest.php]: add comment ``@return array``

**Comment by itkg-alavieille** [comment]
:+1: with checkstyle

**Pull request merged**